### ISSUE TITLE
Clear pending rejection after an Approval action

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-server#15107]
Per a Slack discussion with @kewisch, the PRD didn't cover the case of an Approval action when there's a pending rejection (the PRD only mentions Confirm Approval, my bad).

An Approval should approve the latest version, disable the versions that were pending rejection, and clear the pending rejection. Only the approval email should be sent.

**Comment by diox** [comment]
I really don't like the idea of disabling anything in an approval action.  To avoid surprises, I'm not going to do this:
> disable the versions that were pending rejection

Instead, I'll let it be auto-rejected normally. When the latest version of an add-on is approved (and not pending rejection), all older versions pending rejections will eventually be disabled normally when the deadline is up, and because there has been a version uploaded since, the notification from #14814 won't be sent.

**Comment by jvillalobos** [comment]
> I really don't like the idea of disabling anything in an approval action

What don't you like about it? It will happen eventually, anyway, because of that action.

The problem I have with waiting is that the add-on will end up in a weird state, visible to reviewers and developers, where the latest version is okay but some old versions are pending rejection. An theoretically this could extend to multiple ranges of versions in this state.

**Comment by diox** [comment]
My main issues are that 1) I'm not a big fan of adding more ways for a file status to change than we have already and 2) The reviewer tools actions are already incredibly complex and full of subtle nuances. Forcing the schedule to be obeyed makes it less surprising to developers, reviewers and... AMO engineers.

**Comment by jvillalobos** [comment]
I disagree that it's less surprising for developers. If you get a notification that your review has been resolved, there's a reasonable expectation that all actions will be taken then. Seeing your version status change a week or more later with no notice _is_ surprising. It's also surprising to reviewers since they now have to keep in mind that there are things happening in older versions that change on their own after a delay. There are multiple reviewers handling hundreds of add-ons and with different levels of expertise, so they can't be blamed for being confused when they run into something like this.

That said, we can wait and see if it becomes a real problem in practice.

**Comment by AlexandraMoga** [comment]
I've verified this issue with an add-on that had multiple version with an ongoing pending rejection, including the latest one. The latest version was also Awaiting review, so I approved it manually. The pending rejection status was removed from the latest version only, while the previous versions are still pending rejection.
Example: https://reviewers.addons.allizom.org/en-US/reviewers/review/devhub-listed-ext-07-30

@diox I believe this is what was finally implemented, no?
